aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -0,0 +1,922 @@
+        let old_voter = stake::get_delegated_voter(pool_address);
+        stake::set_delegated_voter_with_cap(&staking_contract.owner_cap, new_voter);
+
+        emit_event(
+            &mut store.update_voter_events,
+            UpdateVoterEvent { operator, pool_address, old_voter, new_voter },
+        );
+    }
+
+    /// Convenient function to allow the staker to reset their stake pool's lockup period to start now.
+    public entry fun reset_lockup(staker: &signer, operator: address) acquires Store {
+        let staker_address = signer::address_of(staker);
+        assert_staking_contract_exists(staker_address, operator);
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+        let pool_address = staking_contract.pool_address;
+        stake::increase_lockup_with_cap(&staking_contract.owner_cap);
+
+        emit_event(&mut store.reset_lockup_events, ResetLockupEvent { operator, pool_address });
+    }
+
+    /// Unlock commission amount from the stake pool. Operator needs to wait for the amount to become withdrawable
+    /// at the end of the stake pool's lockup period before they can actually can withdraw_commission.
+    public entry fun request_commission(operator_account: &signer, staker: address) acquires Store {
+        let operator = signer::address_of(operator_account);
+        assert_staking_contract_exists(staker, operator);
+
+        let store = borrow_global_mut<Store>(staker);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+        assert!(staking_contract.commission_percentage > 0, error::invalid_argument(EZERO_COMMISSION_RATE));
+
+        // Force distribution of any already inactive stake.
+        distribute_internal(staker, operator, staking_contract, &mut store.distribute_events);
+
+        request_commission_internal(
+            operator,
+            staking_contract,
+            &mut store.add_distribution_events,
+            &mut store.request_commission_events,
+        );
+    }
+
+    fun request_commission_internal(
+        operator: address,
+        staking_contract: &mut StakingContract,
+        add_distribution_events: &mut EventHandle<AddDistributionEvent>,
+        request_commission_events: &mut EventHandle<RequestCommissionEvent>,
+    ): u64 {
+        // Unlock just the commission portion from the stake pool.
+        let (total_active_stake, accumulated_rewards, commission_amount) =
+            get_staking_contract_amounts_internal(staking_contract);
+        // Short-circuit if there's no commission to pay.
+        if (commission_amount == 0) {
+            return 0
+        };
+
+        // Add a distribution for the operator.
+        add_distribution(operator, staking_contract, operator, commission_amount, add_distribution_events);
+
+        staking_contract.principal = total_active_stake - commission_amount;
```

Comment:
why does the principal go down if commission is taken?

Reply:
It's not going down. It's updated to be the latest total active stake (inclusive of rewards), excluding commission that's being paid out.